keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -142,6 +157,39 @@ class ApplicationService(
     )
   }
 
+  /**
+   * Adds details about any stateless constraints in the given environment to the [ArtifactSummaryInEnvironment].
+   */
+  private fun addStatelessConstraintSummaries(
+    artifactSummaryInEnvironment: ArtifactSummaryInEnvironment,
+    deliveryConfig: DeliveryConfig,
+    environment: Environment,
+    version: String,
+    artifact: DeliveryArtifact
+  ): ArtifactSummaryInEnvironment {
+    val statelessConstraints: List<StatelessConstraintSummary?> = environment.constraints.filter { constraint ->
```

Comment:
Why the `?` for the type of the list items? There shouldn't be any nulls in the list, right?

Reply:
the `find` can still return null, so the compiler is asking me to play it safe